src/components/views/rooms/NotificationBadge/StatelessNotificationBadge.tsx

```diff
@@ -56,15 +59,27 @@ export const StatelessNotificationBadge = forwardRef<HTMLDivElement, XOR<Props,
             symbol = formatCount(count);
         }
 
+        // We show a dot if either:
+        // * The props force us to, or
+        // * It's just an activity-level notification or (in theory) lower and the room isn't knocked
+        const badgeType =
+            forceDot || (level <= NotificationLevel.Activity && !knocked)
+                ? "dot"
+                : !symbol || symbol.length < 3
```

Comment:
previously, if symbol was falsey, we would not set any of these three classes. We're now setting `badge_2char`. Is that a deliberate change?

Reply:
Hmm, well this was taking your suggestion from https://github.com/matrix-org/matrix-react-sdk/pull/12254#discussion_r1515237609 so I assume not?

Reply:
Oh, actually this is the PR not being split out properly. This is because it needed to be a badge when there was no symbol with mark as unread (because we want that to be badge-sized). I'll try to extract that from this PR.

Reply:
Okay, I've backed this change out from this PR.